### ISSUE TITLE
Terra Select dropdown detaches from select on component resize

### ISSUE THREAD

**Issue by hegedusm** [cerner/terra-core#3478]
# Bug Report

## Description
When the select component resizes and the dropdown is open, the dropdown will detach itself from the component. This happens only if the element does not lose focus during resizing.

## Steps to Reproduce
1.  Open dropdown of a select component
2. Resize the component, but make sure the select still has the focus
3. Dropdown should not be in the correct position, should be slightly off depending where how the component moved during resize.

## Additional Context / Screenshots
![image](https://user-images.githubusercontent.com/28489798/129531045-58bb6ded-eb11-46c6-8bcf-c8826ba61a1c.png)

## Expected Behavior
Hookshot should recalculate position and size on component resize. It currently recalculates only on window resize.

## Possible Solution
If Hookshot https://github.com/cerner/terra-framework/blob/main/packages/terra-hookshot/src/Hookshot.jsx#L213 would listen to parent resize it could recalculate. 

## Environment

* Component Name and Version: Terra Form Select  6.32.0
* Browser Name and Version: Chrome - latest / IE 10
* Node/npm Version: [e.g. Node 8/npm 5]
* Webpack Version:
* Operating System and version (desktop or mobile):

**Comment by ryanthemanuel** [comment]
@hegedusm what is causing the resize to occur? Typically when something resizes like this we've tried to dismiss popup type things.

**Comment by hegedusm** [comment]
@ryanthemanuel We have a consumer that resizes the container using a draggable sidebar. I guess the drag prevents click propagation, but to be honest I am not familiar with the exact implementation.

**Comment by ryanthemanuel** [comment]
@hegedusm this is a defect with how the resize of the draggable sizebar is being handled. We have logged an internal defect for it. Please reach out if you would like to follow this issue and I will give you the JIRA.
